hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionReplicaFragmentVersions.java

```diff
@@ -99,6 +120,10 @@ boolean isDirty() {
         return dirty;
     }
 
+    void markAsSyncRequired(int replicaIndex) {
+        versions[replicaIndex - 1] = REQUIRES_SYNC;
+    }
```

Comment:
Maybe not that important but I always wondered if we should `dirty = true;` in this method? Because we mark dirty in update if version is `REQUIRES_SYNC`.